docs/book/src/developer/providers/contracts/infra-machine.md

```diff
@@ -1,118 +1,486 @@
+| [InfraMachine: initialization completed]                            | Yes       |                                       |
+| [InfraMachine: conditions]                                          | No        |                                       |
+| [InfraMachine: terminal failures]                                   | No        |                                       |
+| [InfraMachineTemplate, InfraMachineTemplateList resource definition] | No        | Mandatory for ClusterClasses support  |
+| [InfraMachineTemplate: support for SSA dry run]                     | No        | Mandatory for ClusterClasses support  |
+| [Multi tenancy]                                                     | No        | Mandatory for clusterctl CLI support  |
+| [Clusterctl support]                                                | No        | Mandatory for clusterctl CLI support  |
+
+Note:
+- `All resources` refers to all the provider's resources "core" Cluster API interacts with;
+  In the context of this page: `InfraMachine`, `InfraMachineTemplate` and corresponding list types
+
+### All resources: scope
+
+All resources MUST be namespace-scoped.
+
+### All resources: `TypeMeta` and `ObjectMeta` field
+
+All resources MUST have the standard Kubernetes `TypeMeta` and `ObjectMeta` fields.
+
+### All resources: `APIVersion` field value
+
+In Kubernetes `APIVersion` is a combination of API group and version.
+Special consideration MUST applies to both API group and version for all the resources Cluster API interacts with.
+
+#### All resources: API group
+
+The domain for Cluster API resources is `cluster.x-k8s.io`, and infrastructure providers under the Kubernetes SIGS org
+generally use `infrastructure.cluster.x-k8s.io` as API group.
+
+If your provider uses a different API group, you MUST grant full read/write RBAC permissions for resources in your API group
+to the Cluster API core controllers. The canonical way to do so is via a `ClusterRole` resource with the [aggregation label]
+`cluster.x-k8s.io/aggregate-to-manager: "true"`.
+
+The following is an example ClusterRole for a `FooMachine` resource in the `infrastructure.foo.com` API group:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+    name: capi-foo-clusters
+    labels:
+      cluster.x-k8s.io/aggregate-to-manager: "true"
+rules:
+- apiGroups:
+    - infrastructure.foo.com
+  resources:
+    - foomachines
+  verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+- apiGroups:
+    - infrastructure.foo.com
+  resources:
+    - foomachinetemplates
+  verbs:
+    - get
+    - list
+    - patch
+    - update
+    - watch
+```
+
+Note: The write permissions allow the Machine controller to set owner references and labels on the InfraMachine resources;
+write permissions are not used for general mutations of InfraMachine resources, unless specifically required (e.g. when
+using ClusterClass and managed topologies).
```

Comment:
I think this sentence doesn't make sense for InfraMachien / InfraMachineTEmplate like it did for InfraCluster / InfraClusterTemplate

Reply:
Good catch, rephrased, PTAL